### PR TITLE
Backport #304 onto v1.14

### PR DESCRIPTION
**Pull request by jedleggett**
#### Summary of Changes
Backport https://github.com/jito-foundation/jito-solana/pull/304 (omits breaking deprecated argument removal)

### CODE REVIEW

File: validator/src/admin_rpc_service.rs [hunk 1]
```diff
@@ -263,6 +288,37 @@ impl AdminRpc for AdminRpcImpl {
...
+    fn set_block_engine_config(
+        &self,
+        meta: Self::Metadata,
+        auth_service_addr: String,
+        backend_addr: String,
+        trust_packets: bool,
+    ) -> Result<()> {
+        debug!("set_block_engine_config request received");
+
+        if !(backend_addr.contains("http") || backend_addr.is_empty()) {
```
Comment (segfaultdoc) [inline validator/src/admin_rpc_service.rs:300]: what if client supplies a string that passes this check but contains invalid URI chars?

**Review by buffalu — APPROVED** [review]

File: validator/src/main.rs [hunk 2]
```diff
@@ -1839,31 +1841,31 @@ pub fn main() {
...
+                .help("Deprecated: Address of the block engine's authentication service.")
                 .conflicts_with("block_engine_url")
         )
         .arg(
             Arg::with_name("relayer_auth_service_address")
                 .long("relayer-auth-service-address")
                 .value_name("relayer_auth_service_address")
                 .takes_value(true)
-                .help("Address of the block engine's authentication service.")
+                .help("Deprecated: Address of the block engine's authentication service.")
```
Comment (buffalu) [inline validator/src/main.rs:1860]: wanna add deprecated, please use relayer_url?

File: validator/src/main.rs [hunk 3]
```diff
@@ -1839,31 +1841,31 @@ pub fn main() {
                 .long("block-engine-address")
                 .value_name("block_engine_address")
                 .takes_value(true)
-                .help("Address of the block engine")
+                .help("Deprecated: Address of the block engine's grpc.")
```
Comment (buffalu) [inline validator/src/main.rs:1844]: wanna add deprecated, please use block_engine_url?

File: validator/src/main.rs [hunk 4]
```diff
@@ -2721,111 +2875,81 @@ pub fn main() {
...
+                    value_of(&matches, "block_engine_auth_service_address")
+                        .expect("couldn't parse block_engine_auth_service_address");
+                block_engine_config.backend_addr = value_of(&matches, "block_engine_address")
+                    .expect("couldn't parse block_engine_address");
+            }
+            (false, false) => {}
+            _ => {
+                eprintln!("Specifying seperate auth and backend addresses for block engine is deprecated.  Recommended to use --block_engine_url instead.\
+                    If using block_engine_auth_service_address and block_engine_address, they must both be provided.");
+                exit(1);
```
Comment (buffalu) [inline validator/src/main.rs:2903]: i don't think we want to exit here for v1.14

Comment (buffalu) [reply validator/src/main.rs:2903]: err actually, nvmd

**Review by buffalu — APPROVED** [review]

File: validator/src/admin_rpc_service.rs (hunk 1 above)
Comment (jedleggett) [reply validator/src/admin_rpc_service.rs]: updated validation functions and used here as suggested

**Review by segfaultdoc — APPROVED** [review]

**Comment by jedleggett** [comment]
Had to revert a clippy suggestion that broke buildkite nits.

**Review by buffalu — APPROVED** [review]

**Pull request merged**